src/main/java/org/elasticsearch/discovery/srv/SrvUnicastHostsProvider.java

```diff
@@ -122,6 +122,13 @@ protected Resolver buildResolver(Settings settings) {
                 logger.warn("Could not create resolver. Using default resolver.", e);
             }
         }
+
+        parent_resolver = parent_resolver == null ? Lookup.getDefaultResolver() : parent_resolver;
```

Comment:
Does getDefaultResolver return a new resolver instance or a shared one?

In other words, will `setTCP(true)` at https://github.com/github/elasticsearch-srv-discovery/pull/15/files#diff-8a2d9bf8eeb5fb563f4a8393fc092565R129 make `useTCP == true` for all resolvers returned by getDefaultResolver?